webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -71,6 +71,14 @@ public DefaultCsvValidationService(final SchoolRepository schoolRepository) {
                 break;
             }
         }
+        if ( csvHeaders.size() > RequiredHeaders.size() ) {
```

Comment:
I'd suggest that you put something before the loop checking the headers. It looks like the existing code will throw if RequiredHeaders > csvHeaders. And now we're adding a condition that there can't be extra headers. So i would suggest, before line 63:
```
    if (csvHeaders.size() != RequiredHeaders.size()) {
        failures.add(...);
        return failures;
    }
```